### ISSUE TITLE
FIX for Asymmetric Transaction Error with ElasticSearch

### ISSUE THREAD

**Issue by phoenix128** [magento/magento2#9930]
`\Magento\Catalog\Model\Product\Gallery\Processor::addImage` does not work with ElasticSearch while importing images programmatically due to undefined index.

### Preconditions

1. Magento CE/EE 2.1.x
2. ElasticSearch configured

### Steps to reproduce

1. Programmatically add a product image using `\Magento\Catalog\Model\Product\Gallery\Processor::addImage`
2. Save the products

### Expected result

1. Product saved with no errors

### Actual result

1. Asymmetric Transaction Error received casue by undefined index "media_type" in `\Magento\Elasticsearch\Model\Adapter\DataMapper\ProductDataMapper::getProductMediaGalleryData` (line 260)

### How to fix:
`\Magento\Catalog\Model\Product\Gallery\Processor::addImage` around line 181:

```
       $mediaGalleryData['images'][] = [
            'file' => $fileName,
            'position' => $position,
            'label' => '',
            'disabled' => (int)$exclude,
        ];
```

Becomes:

```
       $mediaGalleryData['images'][] = [
            'file' => $fileName,
            'position' => $position,
            'label' => '',
            'media_type' => 'image',
            'disabled' => (int)$exclude,
        ];
```

**Comment by phoenix128** [comment]
Created Pull Request: https://github.com/magento/magento2/pull/10610

**Comment by miguelbalparda** [comment]
For a better searchability in the future, can you add the exact error you encountered?

**Comment by phoenix128** [comment]
It is just `Asymmetric Transaction Rollback` while saving. Nothing more.
Tracking down I found it source in the file I preported in my description.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-71801

**Comment by miguelbalparda** [comment]
This should be fixed by #10610

**Comment by miguelbalparda** [comment]
This should be fixed by #10610

**Comment by korostii** [comment]
Hi @miguelbalparda,
#10610 seems to fix this on `develop` branch.
Do you happen to know whether it was fixed for the 2.1.x branch? If it's not then I don't see the point to closing this issue (yet).

By the way, current develop branch seems to represent the future 2.3 version already, so I suspect there might be a need to have that fix back-ported into `2.2.0-preview` separately, as well. 
Could you confirm that somehow? I am not sure if that's correct.
